### DISCUSSION TITLE
DCO Sign-off Required Beginning 10/14/2024

### DISCUSSION THREAD

**Discussion by sebrowne** [trilinos/Trilinos#13497] [Announcements]
With the upcoming move of [Trilinos](https://github.com/trilinos/Trilinos) to a foundation-based governance model (HPSF), we are beginning to require explicit sign-off declarations with the [Developer Certificate of Origin](https://developercertificate.org) in commits that are merged to the Trilinos project.

There is a convenient option to `git commit` that may be used for this purpose: https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s
NOTE: Do not confuse the `-s, --signoff` option with the `-S, --gpg-sign` option!  We do not currently require GPG signing, but we WILL begin requiring a DCO signoff.

Alternatively, one could place the sign-off in a commit message template (https://git-scm.com/docs/git-commit/2.10.5#Documentation/git-commit.txt--tltfilegt)

There is currently a GitHub check that looks at all commits in a pull request and notifies the user if any of them do not contain a DCO sign-off.

![Screenshot 2024-10-01 at 6 41 43 AM](https://github.com/user-attachments/assets/495dd279-6290-42f6-b9c3-b1bb509a9bed)

This check will be changed to be required to pass on Monday, 10/14/2024.

There is a manual override option in that check.  Please do not use it.  Even if you use that manual “Set DCO to pass” button, there is a secondary check (without an override) that will still fail your Pull Request and prevent merging.

**Comment by jhux2** [comment]
Per discussions with @sebrowne, `git commit -s` is sufficient.  The commit message doesn't need the DCO text.
